### DISCUSSION TITLE
Upload inventory file failed

### DISCUSSION THREAD

**Discussion by kunwuliu** [saleweaver/python-amazon-sp-api#632] [Q&A]
**Describe the bug**
An error occured when uploading inventory file
The error:
0		90012	Fatal	The required sku field is missing from the file. To correct this error, download the template again and use the new copy, or insert the required field into your existing file.

Actually I gave it the sku field.

**To Reproduce**

Make a inventory file and use feeds api upload it. 
In my case, I uploaded the inventory file with two fields: sku and add-delete. 
I want to delete the inventory using api.

**Expected behavior**

The related inventory can be deleted

**Desktop (please complete the following information):**

 Linux

**Comment by denisneuf** [comment]
Hi kunwuliu,

I suggest you use the ListingsItems api then you don't need to upload a feed just with the sku you may be able to delete the listing from your inventory:

```
from sp_api.api import ListingsItems
from sp_api.base import SellingApiException
from sp_api.base import Marketplaces
import logging


sku_delete = "4X-5ISB-82J1"
seller_id = "yoursellerid"
issue_locale = "es_ES" 
marketplace_id = "yourmarketplaceid"


try:

    res = ListingsItems(account=store, marketplace=Marketplaces.ES).delete_listings_item(
        sellerId=seller_id,
        sku=sku_delete,
        marketplaceIds=marketplace_id,
        issueLocale=issue_locale
    )

    logging.info(res)
    result = res.payload
    logging.info(result)


except SellingApiException as ex:
    logging.info(ex)
```

Let me know if it works for you.

**Reply by kunwuliu** [reply]
But I can only delete one listing one time in this way. What if I want to delete thousands of listings one time? Is it possible to fix the feeds api?
